### ISSUE TITLE
Failed to route Dialog for Intent name with dot 

### ISSUE THREAD

**Issue by dawwa** [microsoft/BotFramework-Composer#8452]
## To Reproduce

1. Create root bot
2. Create Skill bot
3. Add new Intent with dot in the name, e.g. `My.Test.Intent`
4. Add a new trigger, e.g., OnIntent(My.Test.Intent) to Begin a new dialog, e.g., `MyTestIntentDialog`

Expected Behavior: Route to the `MyTestIntentDialog` properly
Actual Behavior: 
- Route to the `UnknownIntentDialog` dialog even though the `LuisV3 Trace` indicate the correct LUIS Result has returned and recognized the Intent `My.Test.Intent`
- Removing the dot from the intent name makes things back to work.

## Version
2.0

## Browser

- [ ] Electron distribution
- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [X] Windows
- [ ] Ubuntu

**Comment by srinaath** [comment]
@dawwa Could you clarify the steps you did to create an intent name with a period (.) in it. We have validation to specifically block special characters.
![Screen Shot 2021-07-22 at 9 36 05 AM](https://user-images.githubusercontent.com/13004779/126675801-d9dd1726-f0e8-4ba7-b641-a413f12f1879.png)

**Comment by dawwa** [comment]
@srinaath @dmvtech, what I did is just rename an existing intent, and I directly modified from the code, and there is no validation error/warning show up, is the validation only on UI?

**Comment by srinaath** [comment]
Thats right @dawwa the validation is on the UI. I think it makes sense for us to do a validation of the intent name if the schema is directly updated in "Show Code".
